platforms/core-configuration/model-core/src/main/java/org/gradle/internal/reflect/validation/TypeAwareProblemBuilder.java

```diff
@@ -25,8 +25,6 @@
 public interface TypeAwareProblemBuilder extends InternalProblemSpec {
     TypeAwareProblemBuilder withAnnotationType(@Nullable Class<?> classWithAnnotationAttached);
 
-    TypeAwareProblemBuilder typeIsIrrelevantInErrorMessage();
```

Comment:
We can identify which type validation problems should not have the type in the error message based on the problem ID; we no longer need to have a separate field for that.